internal/resource-request-operator/resourceRequest_controller.go

```diff
@@ -41,7 +44,29 @@ func (r *ResourceRequestReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 		return ctrl.Result{}, nil
 	}
 
-	offerErr := r.generateResourceOffer(&resourceRequest)
+	// check if a foreignCluster with an incoming peering
```

Comment:
The comment mentions the incoming peering, but it seems to me this is not checked anywhere (correctly AFAIU). In that case the comment should be rephrased.